### PR TITLE
#171 Interracial Game Rules

### LINKED ISSUE #171: Interracial Game Rules
**Describe your suggestion in full detail below:**
Add the game rules to regular interracial marriages: enable it completely, remove penalties, block it completely.

### PR DESCRIPTION
**Pull request by robmart**
Resolves #171

## Changelog:
* Added the game rule "Interracial Marriage":
    * Unrestricted means you can marry all other races
    * Restricted is the default setting. This setting only makes you able to marry closely related races for example humans and elves
    * Off means that you can't marry other races
* Added the game rule "Interracial Marriage Modifier":
    * Modifier is the default setting. It gives negative modifier to the score when AI is considering a proposed marriage, if it is an interracial marriage
    * No Modifier simply has no modifier to the score 

# How to test
- Use the new gamerules and try different interracial marriages

# Needs to be looked at
* Loc could probably use some polishing

### CODE REVIEW

File: localization/english/wc_game_rules_l_english.yml [hunk 1]
```diff
@@ -4,3 +4,11 @@
  setting_wc_lore_characters_generate_families_on_desc:0 "If vanilla $rule_generate_families$ game rule is enabled, lore characters will be affected by it."
  setting_wc_lore_characters_generate_families_off:0 "Off"
  setting_wc_lore_characters_generate_families_off_desc:0 "If vanilla $rule_generate_families$ game rule is enabled, lore characters will not be affected by it."
+
+ rule_wc_interracial_marriage:0 "#Bold Warcraft#!: Interracial Marriages"
+ setting_wc_interracial_marriage_no_modifier:0 "Unrestricted"
+ setting_wc_interracial_marriage_no_modifier_desc:0 "Interracial marriages will not suffer a negative modifier."
```
Comment (zumbak04) [inline localization/english/wc_game_rules_l_english.yml:10]: Previously unrestricted meant any race can make marriage with every race.
I would bring this option back. This part of the code was responsible for it.
<details>
<summary>Click to expand</summary>

image

</details>

Comment (robmart) [reply localization/english/wc_game_rules_l_english.yml:10]: > 
> 
> Previously unrestricted meant any race can make marriage with every race.
> I would bring this option back. This part of the code was responsible for it.
> Click to expand
> 
> image

Alright, I will make another one, one for no modifier and one for totally unrestricted

File: localization/english/wc_game_rules_l_english.yml [hunk 2]
```diff
@@ -4,3 +4,13 @@
  setting_wc_lore_characters_generate_families_on_desc:0 "If vanilla $rule_generate_families$ game rule is enabled, lore characters will be affected by it."
  setting_wc_lore_characters_generate_families_off:0 "Off"
  setting_wc_lore_characters_generate_families_off_desc:0 "If vanilla $rule_generate_families$ game rule is enabled, lore characters will not be affected by it."
+
+ rule_wc_interracial_marriage:0 "#Bold Warcraft#!: Interracial Marriages"
+ setting_wc_interracial_marriage_unrestricted:0 "Unrestricted"
+ setting_wc_interracial_marriage_unrestricted_desc:0 "All interracial marriages are possible, even those not normally possible."
+ setting_wc_interracial_marriage_no_modifier:0 "Modifier"
+ setting_wc_interracial_marriage_no_modifier_desc:0 "Interracial marriages will not suffer a negative modifier."
```
Comment (zumbak04) [inline localization/english/wc_game_rules_l_english.yml:12]: Sounds more clear.
```suggestion
 setting_wc_interracial_marriage_no_modifier:0 "No Opinion"
 setting_wc_interracial_marriage_no_modifier_desc:0 "Interracial marriages will not suffer a negative marriage opinion."
```

File: localization/english/wc_game_rules_l_english.yml [hunk 3]
```diff
@@ -4,3 +4,13 @@
...
+ setting_wc_interracial_marriage_modifier:0 "Restricted"
```
Comment (zumbak04) [inline localization/english/wc_game_rules_l_english.yml:13]: Default option is highlighted.
```suggestion
 setting_wc_interracial_marriage_modifier:0 "#high Restricted#!"
```

File: localization/english/wc_game_rules_l_english.yml [hunk 4]
```diff
@@ -4,3 +4,13 @@
...
+ setting_wc_interracial_marriage_modifier_desc:0 "Interracial marriages will suffer a negative modifier."
```
Comment (zumbak04) [inline localization/english/wc_game_rules_l_english.yml:14]:
```suggestion
 setting_wc_interracial_marriage_modifier_desc:0 "Interracial marriages will suffer a negative marriage opinion."
```

File: common/game_rules/000_wc_game_rules.txt [hunk 5]
```diff
@@ -3,4 +3,13 @@
 
 	wc_lore_characters_generate_families_on = {  }
 	wc_lore_characters_generate_families_off = {  }
+}
+
+wc_interracial_marriage = {
```
Comment (zumbak04) [inline common/game_rules/000_wc_game_rules.txt:8]: I feel like we need 2 game rules:
One is responsible for the negative marriage opinion.
Another is responsible for the marriage possibility.
Thus you can have all marriages possible, but with a negative opinion.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/game_rules/000_wc_game_rules.txt (hunk 5 above)
Comment (robmart) [reply common/game_rules/000_wc_game_rules.txt:8]: Added in d3b6192

File: localization/english/wc_game_rules_l_english.yml [hunk 6]
```diff
@@ -4,3 +4,18 @@
...
+ setting_wc_interracial_marriage_unrestricted_desc:0 "All interracial marriages are possible, even those not normally possible."
+ setting_wc_interracial_marriage_restricted:0 "#high Restricted#!"
+ setting_wc_interracial_marriage_restricted_desc:0 "Only closely related races can marry."
+ setting_wc_interracial_marriage_off:0 "Off"
+ setting_wc_interracial_marriage_off_desc:0 "Interracial marriages are entirely disabled."
+
+ rule_wc_interracial_marriage_modifier:0 "#Bold Warcraft#!: Interracial Marriages Modifier"
+ setting_wc_interracial_marriage_modifier_no_modifier:0 "No Modifier"
+ setting_wc_interracial_marriage_modifier_no_modifier_desc:0 "Interracial marriages will not suffer a negative marriage opinion."
+ setting_wc_interracial_marriage_modifier_modifier:0 "#high Negative Modifier#!"
```
Comment (zumbak04) [inline localization/english/wc_game_rules_l_english.yml:19]: Sounds more clear.
```suggestion
 rule_wc_interracial_marriage_modifier:0 "#Bold Warcraft#!: Interracial Marriages Opinion"
 setting_wc_interracial_marriage_modifier_no_modifier:0 "No Opinion"
 setting_wc_interracial_marriage_modifier_no_modifier_desc:0 "Interracial marriages will not suffer a negative marriage opinion."
 setting_wc_interracial_marriage_modifier_modifier:0 "#high Negative Opinion#!"
```

File: common/game_rules/000_wc_game_rules.txt [hunk 7]
```diff
@@ -3,4 +3,19 @@
...
+	default = wc_interracial_marriage_restricted
+
+	wc_interracial_marriage_unrestricted = {  }
+	wc_interracial_marriage_restricted = {  }
+	wc_interracial_marriage_off = {  }
+}
+
+wc_interracial_marriage_modifier = {
+	default = wc_interracial_marriage_modifier_no_modifier
```
Comment (zumbak04) [inline common/game_rules/000_wc_game_rules.txt:17]: Isn't `wc_interracial_marriage_modifier_modifier` default?

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: common/game_rules/000_wc_game_rules.txt (hunk 7 above)
Comment (robmart) [reply common/game_rules/000_wc_game_rules.txt:17]: > 
> 
> Isn't `wc_interracial_marriage_modifier_modifier` default?

Whoops, my bad

**Review by zumbak04 — APPROVED** [review]
LGTM! But update the changelog and request @Warcraft-GoA-Development-Team/testers review.

**Review by TheDrPibb1999 — APPROVED** [review]
all looks good to me including loc!

**Review by arithon — APPROVED** [review]
loc LGTM

**Review by hamjustine — APPROVED** [review]

**Pull request merged**